src/SimpleSerialShell.cpp

```diff
@@ -21,30 +25,69 @@ SimpleSerialShell::Command * SimpleSerialShell::firstCommand = NULL;
 class SimpleSerialShell::Command {
     public:
         Command(const __FlashStringHelper * n, CommandFunction f):
-            name(n), myFunc(f) {};
+            nameAndDocs(n), myFunc(f) {};
 
         int execute(int argc, char **argv)
         {
             return myFunc(argc, argv);
         };
 
-        // to sort commands
+        // Comparison used for sort commands
         int compare(const Command * other) const
         {
-            const String otherNameString(other->name);
+            const String otherNameString(other->nameAndDocs);
             return compareName(otherNameString.c_str());
         };
 
         int compareName(const char * aName) const
-        {
-            const String myNameString(name);
-            int comparison = strncasecmp(myNameString.c_str(), aName, SIMPLE_SERIAL_SHELL_BUFSIZE);
-            return comparison;
+        {   
+            // Look for the documentation delimiter and make sure we don't 
+            // consider anything beyond it in the comparison.  
+            //
+            // Note for future consideration: The temporary String here could
+            // be eliminated here by leveraging strlen_P, pgm_read_byte, and 
+            // strncasecmp_P.  That will take a bit of research since 
+            // the header file may have a different name on ESP2886/ESP32.
+            String work(nameAndDocs);
+            int compareLength = SIMPLE_SERIAL_SHELL_BUFSIZE;
+            for (unsigned int i = 0; i < work.length(); i++) {
```

Comment:
FYI an alternative could be to use String's built-in search: `work.indexOf(DOC_DELIMITER)` will find the delimiter for you, or return -1.

That said, this is still easy to understand.